### ISSUE TITLE
dynamo reconfigures logging underneath user

### ISSUE THREAD

**Issue by wconstab** [pytorch/torchdynamo#517]
It appears that after running torchdynamo, the behavior of logging is altered.  I wouldn't expect this from a client perspective.

In the repro below, only the following prints: `INFO:__main__:Hello`, but never "Goodbye"

Torchdynamo does configure the logger extensively:
https://github.com/pytorch/torchdynamo/blob/main/torchdynamo/utils.py#L39

So i'm not sure what the best fix is, to balance user considerations with whatever customizations torchdynamo wants to make.

```
import torch
import torchdynamo
import logging

logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)

def toy_example(a, b):
    return torch.abs(a) + 1

logger.info("Hello")
with torchdynamo.optimize("eager"):
    for _ in range(10):
        toy_example(torch.randn(10), torch.randn(10))

logger.info("Goodbye")
```

**Comment by voznesenskym** [comment]
I can take a look, I'm relatively
blocked on other stuff today.

**Comment by voznesenskym** [comment]
One more example, ... and @wconstab there's actually a more malicious pattern here, check this out:

```
import torch
import torchdynamo
import logging

logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)

def toy_example(a, b):
    return torch.abs(a) + 1

logger.info("Hello")
with torchdynamo.optimize("eager"):
    for _ in range(10):
        toy_example(torch.randn(10), torch.randn(10))

logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)

logger.info("Goodbye")
```

This also does not print goodbye! Making for a rather poor UX.

Now, I have a fix which makes the example in this comment work, however, I have not figured out a fix to make your example print `Goodbye` correctly. Would we be interested in landing a partial fix?

**Comment by jansel** [comment]
This is coming from here: https://github.com/pytorch/torchdynamo/blob/9630624b5a870566786a991dc653edb7d9e5a6d0/torchdynamo/utils.py#L70

**Comment by voznesenskym** [comment]
> This is coming from here: https://github.com/pytorch/torchdynamo/blob/9630624b5a870566786a991dc653edb7d9e5a6d0/torchdynamo/utils.py#L70

Yes. Calling shutdown and importlib reloading logging partially fixes it, but I've noticed found a way to restore the prior logging config.

**Comment by jansel** [comment]
I think we just need to set `"disable_existing_loggers": False` in the config dictionary.

**Comment by mlazos** [comment]
This was already fixed by #367
